src/app.ts

```diff
@@ -0,0 +1,35 @@
+import { ApolloServer } from 'apollo-server';
+import { Request } from 'express';
+import { typeDefs } from './schema';
+import { resolvers } from './resolvers/mainResolver';
+import { logger } from './logger';
+import { authenticationContextMiddleware } from './middleware/authentication';
+import { loggingContextMiddleware } from './middleware/logging';
+import { BooksDataSource } from './dataSources/booksDataSource';
+import { CustomersDataSource } from './dataSources/customersDataSource';
+
+const booksDataSource = new BooksDataSource();
+const customersDataSource = new CustomersDataSource();
+
+const createContext = ({ req }: { req: Request }) => {
+  // Initialize an empty context object
+  let context = {};
+
+  // Apply authentication and logging middleware
+  context = { ...context, ...authenticationContextMiddleware({ req }) };
+  context = { ...context, ...loggingContextMiddleware({ req }) };
```

Comment:
We can safely create the middleware at once:
```suggestion
  context = { ...authenticationContextMiddleware({ req }), ...loggingContextMiddleware({ req }) };
```
And we don't even need to spread `context` here, is it will always be an empty object {} at this point.
We can even return the whole context with a one liner, but for the sake of readability let's keep it like that

Reply:
You are right, but now I added the dataSources to the context as well, therefore I think it is more readable on separate lines.